test/unit/lib/plugins/aws/deployFunction.test.js

```diff
@@ -548,3 +549,256 @@ describe('AwsDeployFunction', () => {
     });
   });
 });
+
+describe('AwsDeployFunction - runServerless', () => {
+  describe('when ecr is used', () => {
+    const awsRequestStubMap = {
+      Lambda: {
+        getFunction: {
+          Configuration: {
+            LastModified: '2020-05-20T15:34:16.494+0000',
+          },
+        },
+      },
+    };
+
+    it('should fail if `functions[].image` references image without path and uri', async () => {
+      await expect(
+        runServerless({
+          fixture: 'function',
+          cliArgs: ['deploy', 'function', '-f', 'fnProviderInvalidImage'],
+          awsRequestStubMap,
+          configExt: {
+            provider: {
+              ecr: {
+                images: {
+                  invalidimage: {},
+                },
+              },
+            },
+            functions: {
+              fnProviderInvalidImage: {
+                image: 'invalidimage',
+              },
+            },
+          },
+        })
+      ).to.be.rejectedWith(
+        'Either "uri" or "path" property needs to be set on image "invalidimage"'
```

Comment:
Let's rely on codes, as e.g. here: https://github.com/serverless/serverless/blob/30015eafd2fb9d2e82d8f34ee8f10c1fb4e536a0/test/unit/lib/plugins/aws/package/compile/events/httpApi.test.js#L456-L458

(Human readable message are subject to typo's, grammar improvements)

Reply:
good call :+1: